core/src/main/java/org/springframework/security/authorization/AuthorizationObservationContext.java

```diff
@@ -69,18 +69,18 @@ public T getObject() {
 	}
 
 	/**
-	 * Get the observed {@link AuthorizationDecision}
-	 * @return the observed {@link AuthorizationDecision}
+	 * Get the observed {@link AuthorizationResult}
+	 * @return the observed {@link AuthorizationResult}
 	 */
-	public AuthorizationDecision getDecision() {
+	public AuthorizationResult getDecision() {
 		return this.decision;
 	}
 
 	/**
-	 * Set the observed {@link AuthorizationDecision}
-	 * @param decision the observed {@link AuthorizationDecision}
+	 * Set the observed {@link AuthorizationResult}
+	 * @param decision the observed {@link AuthorizationResult}
 	 */
-	public void setDecision(AuthorizationDecision decision) {
+	public void setDecision(AuthorizationResult decision) {
```

Comment:
To remain [binary compatible](https://wiki.eclipse.org/Evolving_Java-based_APIs_2#Evolving_API_interfaces_-_API_methods), we cannot change return types. Instead, please add a new setter (called `setAuthorizationResult`) and deprecate this setter.